src/traces/contour/find_all_paths.js

```diff
@@ -84,13 +84,13 @@ function makePath(pi, loc, edgeflag) {
         if(equalPts(pts[pts.length - 1], pts[pts.length - 2])) pts.pop();
         locStr = loc.join(',');
 
+        var atEdge = (marchStep[0] && (loc[0] < 0 || loc[0] > n - 2)) ||
+                (marchStep[1] && (loc[1] < 0 || loc[1] > m - 2)),
+            closedLoop = (locStr === startLocStr) && (marchStep.join(',') === startStepStr);
+
         // have we completed a loop, or reached an edge?
-        if((locStr === startLocStr && marchStep.join(',') === startStepStr) ||
-            (edgeflag && (
-                (marchStep[0] && (loc[0] < 0 || loc[0] > n - 2)) ||
-                (marchStep[1] && (loc[1] < 0 || loc[1] > m - 2))))) {
-            break;
-        }
+        if((closedLoop) || (edgeflag && atEdge)) break;
```

Comment:
Actually, if we're worried about perf, I should get rid of all those `str = pair.join(',')` statements - they make the comparison code shorter, but it's way faster to just compare integers twice. Since `marchStep.join(',')` is in the first clause it would save almost nothing to revert.
https://github.com/plotly/plotly.js/issues/1311